sdk/tables/azure-data-tables/migration_guide.md

```diff
@@ -0,0 +1,298 @@
+
+## Migration benefits
+
+As Azure has matured and been embraced by a more diverse group of developers, we have been focused on learning the patterns and practices to best support developer productivity and to understand the gaps that the Python client libraries have.
+
+There were several areas of consistent feedback expressed across the Azure client library ecosystem. One of the most important is that the client libraries for different Azure services have not had a consistent approach to organization, naming, and API structure. Additionally, many developers have felt that the learning curve was difficult, and the APIs did not offer a good, approachable, and consistent onboarding story for those learning Azure or exploring a specific Azure service.
+
+To improve the development experience across Azure services, a set of uniform [design guidelines](https://azure.github.io/azure-sdk/general_introduction.html) were created for all languages to drive a consistent experience with established API patterns for all services. A set of [Python guidelines](https://azure.github.io/azure-sdk/python_design.html) were also introduced to ensure that Python clients have a natural and idiomatic feel with respect to the Python ecosystem. The new `azure-data-tables` follows these guidelines.
+
+### Cross Service SDK improvements
+
+The modern `azure-data-tables` client library also provides the ability to share in some of the cross-service improvements made to the Azure development experience, such as:
+- A unified logging and diagnostics pipeline offering a common view of the activities across each of the client libraries
+
+### Performance improvements
+
+Use this section to advertise the performance improvements in new package when compared to the old one. Skip this section if no perf improvements are found yet.
+@annatisch, should we remove this section?
+
+### New features
+
+We have a variety of new features available in the new Azure Data Tables library:
+* Ability to submit a batch of operations with the `TableClient.submit_transaction` method
+* Separates Table and Entity level operations into two clients, `TableServiceClient` and `TableClient`
+* Adds the ability to query tables and list entities explicitly.
+
+## Important changes
+
+### Package names and namespaces
+
+The package name has been changed from `azure-cosmosdb-table` to `azure-data-tables`. This package can be used to access and manipulate data stored in a CosmosDB Table account or an Azure Storage Tables account.
+
+### Client hierarchy and constructors
+
+In the interest of simplicity, there are only two clients, `TableServiceClient` for account-level interactions and `TableClient` for table-level interactions. This is in contrast to the single `ServiceClient` used for interacting with the account level operations and entity level operations. The two clients will simplify the method calls by not requiring a Table name to be included on each service call.
```

Comment:
I feel this line needs rephrasing. It's a bit contradictory to say "in the interest of simplicity there are only two clients" when that is in fact an increase in complexity by a factor of 2 ;)
How about something more like, "In order to provide an improved conceptual experience, we have divided the single `ServiceClient` into two clients; `TableServiceClient` and `TableClient`. The former provides access to all account-level operations..." etc etc